dex/networks/btc/script.go

```diff
@@ -180,23 +191,35 @@ const (
 	//   - 22 bytes P2PKH output script
 	P2WPKHOutputSize = TxOutOverhead + P2WPKHPkScriptSize // 31
 
-	// MimimumTxOverhead is the size of an empty transaction.
+	// MinimumTxOverhead is the size of an empty transaction.
 	// 4 bytes version + 4 bytes locktime + 2 bytes of varints for the number of
 	// transaction inputs and outputs
-	MimimumTxOverhead = 4 + 4 + 1 + 1 // 10
+	MinimumTxOverhead = 4 + 4 + 1 + 1 // 10
 
 	// InitTxSizeBase is the size of a standard serialized atomic swap
 	// initialization transaction with one change output and no inputs. This is
 	// MsgTx overhead + 1 P2PKH change output + 1 P2SH contract output. However,
 	// the change output might be P2WPKH, in which case it would be smaller.
-	InitTxSizeBase = MimimumTxOverhead + P2PKHOutputSize + P2SHOutputSize // 10 + 34 + 32 = 76
+	InitTxSizeBase = MinimumTxOverhead + P2PKHOutputSize + P2SHOutputSize // 10 + 34 + 32 = 76
 	// leaner with P2WPKH+P2SH outputs: 10 + 31 + 32 = 73
 
 	// InitTxSize is InitTxBaseSize + 1 P2PKH input
 	InitTxSize = InitTxSizeBase + RedeemP2PKHInputSize // 76 + 149 = 225
 	// Varies greatly with some other input types, e.g nested witness (p2sh with
 	// p2wpkh redeem script): 23 byte scriptSig + 108 byte (75 vbyte) witness = ~50
 
+	// InitTxSizeBaseSegwit is the size of a standard serialized atomic swap
+	// initialization transaction with one change output and no inputs. The
+	// change output is assumed to be segwit. 10 + 31 + 43 = 84
+	InitTxSizeBaseSegwit = MinimumTxOverhead + P2WPKHOutputSize + P2WSHOutputSize
+
+	// InitTxSizeSegwit is InitTxSizeSegwit + 1 P2WPKH input.
+	// 84 vbytes base tx
+	// 41 vbytes base tx input
+	// 109wu witness +  2wu segwit marker and flag = 28 vbytes
+	// total = 153 vbytes
+	InitTxSizeSegwit = InitTxSizeBaseSegwit + RedeemP2WPKHInputSize + ((RedeemP2WPKHInputWitnessWeight + 2 + 3) / 4)
```

Comment:
We have to add the 2 marker and flag weight units and take the final vsize calculation for the transaction at the same time. We can't, say, assign a variable `P2WPHKInputVBytes =  RedeemP2WPKHInputSize + ((RedeemP2WPKHInputWitnessWeight + 2 + 3) / 4)`, because if there were multiple inputs, it would count the marker and flag for each input. We also can't create `P2WPHKInputVBytes` without the marker and flag bytes. because the bytes must be added in before rounding.